tests/pytest/pems/districts/test_views.py

```diff
@@ -1,21 +1,33 @@
 import pytest
 
+from django.urls import reverse
 from pems.districts import views
+from pems.districts.models import District
 
 
-class TestDistrictView:
+class TestIndexView:
     @pytest.fixture
     def view(app_request):
-        v = views.DistrictView()
-        v.setup(app_request, district=1)
+        v = views.IndexView()
+        v.setup(app_request)
 
         return v
 
+    @pytest.mark.django_db
     def test_get_context_data(self, view):
 
         context = view.get_context_data()
 
-        assert context["district"] == 1
+        assert set(context.get("districts").get("all")) == set(District.objects.all())
 
     def test_template_name(self, view):
-        assert view.template_name == "districts/district.html"
+        assert view.template_name == "districts/index.html"
+
+
+@pytest.mark.django_db
+def test_district_view(client, model_District):
+    url = reverse("districts:district", kwargs={"district_number": 1})
```

Comment:
This is great, much more readable now that everything has its own name :+1: 

One small note here, it might be a little more future proof to pass `model_District.number` here instead of hardcoding to `1`. That way if the fixture ever changes, you don't have to worry as much about the test failing.

Reply:
For sure 👍, made the change in 6f2679caf23d30d109b4ccd4cef74e55e7c5d492